### ISSUE TITLE
Metallb controller/speaker fails to start on k8s 1.27.0; upgrade of client-go required

### ISSUE THREAD

**Issue by bh-tt** [metallb/metallb#1893]
On startup metallb tries to retrieve all API resources, and it crashes with a panic nil pointer dereference. I cannot pinpoint the specific resource/CRD causing this, however upgrading to client-go 0.27.0 seems to fix the issue (which also occurs with tools like k9s). 

Could you perhaps release a new version with client-go upgraded to 0.27.0?

Related log from controller (speaker pods fails with a similar error):
```
{"branch":"v0.13.9","caller":"main.go:155","commit":"4b41fd5","goversion":"gc / go1.19.7 / amd64","level":"info","msg":"MetalLB controller starting version 0.13.9 (commit 4b41fd5, branch v0.13.9)","ts":"2023-04-13T14:38:20Z","version":"0.13.9"} 
panic: runtime error: invalid memory address or nil pointer dereference                                                                                                                                                                              
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1281a50]                                                                                                                                                                              
                                                                                                                                                                                                                                                     
goroutine 1 [running]:                                                                                                                                                                                                                               
k8s.io/client-go/discovery.convertAPIResource(...)                                                                                                                                                                                                   
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/k8s.io/client-go@v0.26.0/discovery/aggregated_discovery.go:88                                                                                                                                  
k8s.io/client-go/discovery.convertAPIGroup({{{0x0, 0x0}, {0x0, 0x0}}, {{0xc0005266c0, 0x15}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})                                                                                                                     
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/k8s.io/client-go@v0.26.0/discovery/aggregated_discovery.go:69 +0x5f0                                                                                                                           
k8s.io/client-go/discovery.SplitGroupsAndResources({{{0xc00004a138, 0x15}, {0xc00004f2c0, 0x1b}}, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})                                                                                                   
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/k8s.io/client-go@v0.26.0/discovery/aggregated_discovery.go:35 +0x2f8                                                                                                                           
k8s.io/client-go/discovery.(*DiscoveryClient).downloadAPIs(0x1aaaef1?)                                                                                                                                                                               
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/k8s.io/client-go@v0.26.0/discovery/discovery_client.go:310 +0x47c                                                                                                                              
k8s.io/client-go/discovery.(*DiscoveryClient).GroupsAndMaybeResources(0xc0002f8960?)                                                                                                                                                                 
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/k8s.io/client-go@v0.26.0/discovery/discovery_client.go:198 +0x5c                                                                                                                               
k8s.io/client-go/discovery.ServerGroupsAndResources({0x1d35e98, 0xc000268600})                                                                                                                                                                       
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/k8s.io/client-go@v0.26.0/discovery/discovery_client.go:392 +0x59                                                                                                                               
k8s.io/client-go/discovery.(*DiscoveryClient).ServerGroupsAndResources.func1()                                                                                                                                                                       
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/k8s.io/client-go@v0.26.0/discovery/discovery_client.go:356 +0x25                                                                                                                               
k8s.io/client-go/discovery.withRetries(0x2, 0xc0005e2e20)                                                                                                                                                                                            
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/k8s.io/client-go@v0.26.0/discovery/discovery_client.go:621 +0x72                                                                                                                               
k8s.io/client-go/discovery.(*DiscoveryClient).ServerGroupsAndResources(0x0?)                                                                                                                                                                         
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/k8s.io/client-go@v0.26.0/discovery/discovery_client.go:355 +0x3a                                                                                                                               
k8s.io/client-go/restmapper.GetAPIGroupResources({0x1d35e98?, 0xc000268600?})                                                                                                                                                                        
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/k8s.io/client-go@v0.26.0/restmapper/discovery.go:148 +0x42                                                                                                                                     
sigs.k8s.io/controller-runtime/pkg/client/apiutil.NewDynamicRESTMapper.func1()                                                                                                                                                                       
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/client/apiutil/dynamicrestmapper.go:86 +0x25                                                                                                        
sigs.k8s.io/controller-runtime/pkg/client/apiutil.(*dynamicRESTMapper).setStaticMapper(...)                                                                                                                                                          
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/client/apiutil/dynamicrestmapper.go:119                                                                                                             
sigs.k8s.io/controller-runtime/pkg/client/apiutil.NewDynamicRESTMapper(0xc0004c03f0?, {0x0, 0x0, 0x5abdc44c9b41d01?})                                                                                                                                
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/client/apiutil/dynamicrestmapper.go:99 +0x176                                                                                                       
sigs.k8s.io/controller-runtime/pkg/cluster.setOptionsDefaults.func1(0xc0004cd2d0?)                                                                                                                                                                   
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/cluster/cluster.go:217 +0x25                                                                                                                        
sigs.k8s.io/controller-runtime/pkg/cluster.New(0xc000354480, {0xc00015d7a8, 0x1, 0x0?})                                                                                                                                                              
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/cluster/cluster.go:159 +0x18d                                                                                                                       
sigs.k8s.io/controller-runtime/pkg/manager.New(_, {0xc0004cd2d0, 0x0, 0x0, {{0x1d31548, 0xc00040f6c0}, 0x0}, 0x0, {0x0, 0x0}, ...})                                                                                                                  
    /bitnami/blacksmith-sandox/metallb-0.13.9/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/manager/manager.go:351 +0xf9                                                                                                                        
go.universe.tf/metallb/internal/k8s.New(0xc0006b6dd0)                                                                                                                                                                                                
    /bitnami/blacksmith-sandox/metallb-0.13.9/src/github.com/metallb/metallb/internal/k8s/k8s.go:129 +0x69d                                                                                                                                          
main.main()                                                                                                                                                                                                                                          
    /bitnami/blacksmith-sandox/metallb-0.13.9/src/github.com/metallb/metallb/controller/main.go:207 +0x10ae 
```

Related commit: https://github.com/kubernetes/client-go/commit/2bd557faec655a441ac0927c41cebdabf0c315a8

**Comment by SammyA** [comment]
https://github.com/kubernetes/kubernetes/pull/116666

The fix from client-go v0.27.x was backported to v0.26.4 , which might be less scary to upgrade.

**Comment by fedepaol** [comment]
Thanks @SammyA ! Moreover, I think client go 0.27 is incompatible with the latest controller-runtime, so we need to wait. I am gonna file a pr to bump to the minor.
